ferrocene/ci/scripts/detect-unused-cargo-patches.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+import subprocess
+import tomllib
+from collections import defaultdict
+
+
+def find_files(name):
+    cmd = subprocess.run(
+        ["git", "ls-tree", "HEAD", "--name-only", "-r", "-z"],
+        stdout=subprocess.PIPE,
+        check=True,
+        text=True,
+    )
+
+    # The -z flag changes the output to separate file names with a null byte, preventing files with
+    # spaces or newlines from messing things up.
+    for file in cmd.stdout.split("\0"):
+        if file == name or file.endswith(f"/{name}"):
+            yield file
+
+
+def find_unused_patches(file):
+    with open(file, "rb") as f:
+        try:
+            toml = tomllib.load(f)
+        # Probably due to conflict markers:
```

Comment:
If conflict markers are present, this probably shouldn't pass, right?

Reply:
Conflict markers will already be caught by the step right before, so there is no risk of CI letting files with conflict markers go through. I chose not to make this an error if someone has other files called Cargo.lock that are not valid TOML for whatever reason. I can switch it to an error if you'd prefer.